src/sentry/api/endpoints/organization_events_stats.py

```diff
@@ -253,7 +253,7 @@ def get_event_stats(
                 use_metrics_layer=force_metrics_layer
                 or batch_features.get("organizations:use-metrics-layer", False),
                 on_demand_metrics_enabled=use_on_demand_metrics
-                and batch_features.get("organizations:on-demand-metrics-extraction", False),
+                and batch_features.get("organizations:on-demand-metrics-extraction-widgets", False),
```

Comment:
should be `or` since alerts / widgets might diverge but the existence of either means on-demand can be tried